test/controllers/event1/member/BenefitsPaidEarlyControllerSpec.scala

```diff
@@ -0,0 +1,138 @@
+
+  private val extraModules: Seq[GuiceableModule] = Seq[GuiceableModule](
+    bind[UserAnswersCacheConnector].toInstance(mockUserAnswersCacheConnector)
+  )
+
+  private val validValue = "abc"
+
+  override def beforeEach: Unit = {
+    super.beforeEach
+    reset(mockUserAnswersCacheConnector)
+  }
+
+  "BenefitsPaidEarly Controller" - {
+
+    "must return OK and the correct view for a GET" in {
+
+      val application = applicationBuilder(userAnswers = Some(emptyUserAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val result = route(application, request).value
+
+        val view = application.injector.instanceOf[BenefitsPaidEarlyView]
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form, waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
+
+      val userAnswers = UserAnswers().set(BenefitsPaidEarlyPage, validValue).success.value
+
+      val application = applicationBuilder(userAnswers = Some(userAnswers)).build()
+
+      running(application) {
+        val request = FakeRequest(GET, getRoute)
+
+        val view = application.injector.instanceOf[BenefitsPaidEarlyView]
+
+        val result = route(application, request).value
+
+        status(result) mustEqual OK
+        contentAsString(result) mustEqual view(form.fill(validValue), waypoints)(request, messages(application)).toString
+      }
+    }
+
+    "must save the answer and redirect to the next page when valid data is submitted" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
+        .thenReturn(Future.successful(()))
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers), extraModules)
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, postRoute).withFormUrlEncodedBody(("value", "true"))
+
+        val result = route(application, request).value
+        val updatedAnswers = emptyUserAnswers.set(BenefitsPaidEarlyPage, validValue).success.value
+
+        status(result) mustEqual SEE_OTHER
+        redirectLocation(result).value mustEqual BenefitsPaidEarlyPage.navigate(waypoints, emptyUserAnswers, updatedAnswers).url
+      }
+    }
+
+    "must return bad request when invalid data is submitted" in {
+      when(mockUserAnswersCacheConnector.save(any(), any(), any())(any(), any()))
+        .thenReturn(Future.successful(()))
+
+      val application =
+        applicationBuilder(userAnswers = Some(emptyUserAnswers), extraModules)
+          .build()
+
+      running(application) {
+        val request =
+          FakeRequest(POST, postRoute).withFormUrlEncodedBody(("value", ""))
+
+        val view = application.injector.instanceOf[BenefitsPaidEarlyView]
+        val boundForm = form.bind(Map("value" -> ""))
```

Comment:
this is not invalid. text area is optional.

you may test with this instead.
val invalidValue ="*" * 151

Reply:
Amended in next push.